Extensions/GoogleOptimize/Src/Tasks/GoogleOptimize/operations/JwtHandler.ts

```diff
@@ -0,0 +1,85 @@
+import * as fs from 'fs'
+import * as crypto from 'crypto'
+import * as tl from 'azure-pipelines-task-lib/task'
+const request = require('request');
+import {TaskParameter} from './../models/TaskParameter';
+let fs = require("fs");
+
+export class JwtHandler{
+    private issuer: string;
+    private audience: string;
+    private scope: string;
+    private privatekey: string;
+    private param: TaskParameter;
+    private authClaimSet = {};
+
+    constructor(){
+        this.param = new TaskParameter() ;
+        let endpointId = this.param.getEndpoint() ;
+        this.issuer = tl.getEndpointAuthorizationParameter(endpointId , 'Issuer' , true);
+        this.audience = tl.getEndpointAuthorizationParameter(endpointId , 'Audience' , true);
+        this.scope = tl.getEndpointAuthorizationParameter(endpointId , 'Scope' , true);
+        this.privatekey = tl.getEndpointAuthorizationParameter(endpointId , 'PrivateKey' , true);
+        this.authClaimSet.iss = this.issuer ;
+        this.authClaimSet.aud = this.audience ;
+        this.authClaimSet.scope = this.scope ;
+    }
+
+    private authHeader = {
+    		'alg': 'RS256',
+    		'typ': 'JWT'
+    };
+
+    private urlEscape(source){
+        return source.replace(/\+/g, '-').replace(/\//g, '_').replace(/\=+$/, '');
+    }
+
+    private base64Encode(obj){
+        var encoded = new Buffer(JSON.stringify(obj), 'utf8').toString('base64');
+    	return this.urlEscape(encoded);
+    }
+
+    public authorize(callback){
+        let self = this
+    	let now = parseInt(Date.now() / 1000, 10); // Google wants us to use seconds
+    	let cipher ;
+    	let	signatureInput ;
+    	let	signatureKey = this.privatekey ;
+        let	signature ;
+    	let	jwt ;
+    	this.authClaimSet.iat = now;
+    	this.authClaimSet.exp = now + 60; // Token valid for one minute
+        signatureKey = signatureKey.replace(/\\n/g, '\n');
```

Comment:
AuthClaimSet is being set, spread across constructor and here. Instead can you have a single private method to getAuthClaimSet?